alert-database/src/main/java/com/synopsys/integration/alert/database/api/StaticJobAccessor.java

```diff
@@ -207,15 +208,22 @@ public StaticJobAccessor(
 
         // TODO running project name pattern checks in java code, try to do this in SQL instead (Won't need to return DistributionJobEntity anymore if this happens
         return distributionJobEntities.stream()
-                   .filter(distributionJobEntity -> !(distributionJobEntity.getBlackDuckJobDetails().getFilterByProject() &&
-                                                          distributionJobEntity.getBlackDuckJobDetails().getProjectNamePattern() != null &&
-                                                          !Pattern.matches(distributionJobEntity.getBlackDuckJobDetails().getProjectNamePattern(), projectName) &&
-                                                          !distributionJobEntity.getBlackDuckJobDetails().getBlackDuckJobProjects().contains(projectName))
-                   )
+                   .filter(distributionJobEntity -> filterByProjects(distributionJobEntity, projectName))
                    .map(this::convertToFilteredDistributionJobResponseModel)
                    .collect(Collectors.toList());
     }
 
+    private boolean filterByProjects(DistributionJobEntity distributionJobEntity, String projectName) {
+        BlackDuckJobDetailsEntity blackDuckJobDetails = distributionJobEntity.getBlackDuckJobDetails();
+        String projectNamePattern = blackDuckJobDetails.getProjectNamePattern();
+        return !blackDuckJobDetails.getFilterByProject() ||
+                   (projectNamePattern != null && Pattern.matches(projectNamePattern, projectName)) ||
+                   blackDuckJobDetails.getBlackDuckJobProjects()
+                       .stream()
+                       .map(BlackDuckJobProjectEntity::getProjectName)
+                       .anyMatch(projectName::equals);
```

Comment:
I still find the readability here pretty low. It might be beneficial to introduce if-statements and short-circuiting.

Reply:
In reality, this should be replaced by a repository implementation that uses postgres keywords to query the DB using projectNamePattern. The query will be much clearer as a lot of this content is what's required to filter the response data a second time. I'd be happy to add short circuits if we think it's necessary

Reply:
I hear you, but this is the implementation we're stuck with for now, so I think it's best to make it as readable as possible.